nibabel/funcs.py

```diff
@@ -224,7 +224,18 @@ def as_closest_canonical(img, enforce_diag=False):
     # we need to transform the data
     arr = img.get_data()
     t_arr = apply_orientation(arr, ornt)
-    return img.__class__(t_arr, out_aff, img.header)
+    # Also apply the transform to the dim_info fields
+    new_hdr = img.header.copy()
+    new_dim = list(new_hdr.get_dim_info())
+    for idx, value in enumerate(new_dim):
```

Comment:
Ah - yes - I remember.   Values are (which axis is frequency, which axis is phase, which axis is slice) where None has the obvious meaning.